lib/utils/dates.ts

```diff
@@ -46,40 +46,48 @@ export const isValidDate = (maybeDate: unknown): boolean => {
   return false;
 };
 
+/**
+ * Get total number of days in a particular month
+ *
+ * @param      {Date}    date    Date of the month that should be considered
+ * @return     {number}  Total number of dates in the particular month
+ */
 export const daysInMonth = (date: Date): number => {
-  return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
+  return new Date(date.getUTCFullYear(), date.getUTCMonth() + 1, 0).getUTCDate();
 };
 
 /**
  * For a given date, get the ISO week number
- * @param date
+ * @param    date
+ * @returns  ISO week number of the specified date
  */
 export const getWeekNumber = (date: Date): number => {
-  const newYear = new Date(date.getFullYear(), 0, 1);
-  let day = newYear.getDay(); // The day of week the year begins on
-  day = day >= 0 ? day : day + 7;
-  const dayNum =
-    Math.floor(
-      (date.getTime() - newYear.getTime() - (date.getTimezoneOffset() - newYear.getTimezoneOffset()) * 60000) /
-        86400000,
-    ) + 1;
-  let weekNum;
-  // If the year starts before the middle of a week
-  if (day < 4) {
-    weekNum = Math.floor((dayNum + day - 1) / 7) + 1;
-    if (weekNum > 52) {
-      const nYear = new Date(date.getFullYear() + 1, 0, 1);
-      let nDay = nYear.getDay();
-      nDay = nDay >= 0 ? nDay : nDay + 7;
-      // If the next year starts before the middle of the week, it is week #1 of that year
-      weekNum = nDay < 4 ? 1 : 53;
-    }
-  } else {
-    weekNum = Math.floor((dayNum + day - 1) / 7);
-  }
-  return weekNum;
+  const d = new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()));
+  d.setUTCDate(d.getUTCDate() + 4 - (d.getUTCDay() || 7));
+  return Math.ceil(((d.getTime() - new Date(Date.UTC(d.getUTCFullYear(), 0, 1)).getTime()) / 864e5 + 1) / 7)
 };
 
+/**
+ * Gets the number of week days in range, including the end date
+ *
+ * @param   {Date}    start      Start date
+ * @param   {Date}    end        End date
+ * @param   {number}  dayOfWeek  Day of week to count
+ * @return  Number of week days in the range, including the end date
+ */
+export const getNumberOfWeekDaysInRange = ({start, end, dayOfWeek = 0}: {start: Date, end: Date, dayOfWeek?: 0 | 1 | 2 | 3 | 4 | 5 | 6}): number => {
```

Comment:
Since this function is only used in tests, not in the main romcal library, could you move it at the top of `dates.test.ts`?

Reply:
You’re right, I move it into `dates.test.ts`.

Note, however, that this function is sourced from [SO](https://stackoverflow.com/a/37408412/3408342). I think we could add a source to the JSDoc (which should be changed to TSDoc) and probably merge all those variables, as they are used only once. Something like this:

```ts
/**
 * Gets the number of week days in range, including the end date
 * 
 * @remarks This function was sourced from [StakOverflow](https://stackoverflow.com/a/37408412/3408342).
 *
 * @param    start     - Start date
 * @param    end       - End date
 * @param    dayOfWeek - Day of week to count
 * @returns  Number of week days in the range, including the end date
 */
const getNumberOfWeekDaysInRange = ({start, end, dayOfWeek = 0}: {start: Date, end: Date, dayOfWeek?: 0 | 1 | 2 | 3 | 4 | 5 | 6}): number => {
  // Calculate the number of weeks (even partial) from the first target day to the end
  return Math.ceil(
    // Calculate the number of days from the first target day to the end
    Math.max(
      // Calculate the number of days between start and end, including the end date (thus `+ 1` at the end)
      Math.ceil((end.getTime() - start.getTime()) / 864e5) + 1
      // Calculate the nb of days before the next target day (e.g. next Sunday after start)
      - (7 + dayOfWeek - start.getUTCDay()) % 7,
      0
    ) / 7
  );
}
```

Reply:
This should be fixed now.